For_Andrey/task_3.py

```diff
@@ -0,0 +1,241 @@
+        else:
+            # For petrol car
+            used_car_cost = Car.NEW_CAR_COST - (Car.DECREASE_VALUE_PETROL_PER_THOUSAND_KM * (self.mileage // 1000))
+        if used_car_cost <= 0:
+            return 0
+        else:
+            return used_car_cost
+
+    def mileage_before_disposal(self) -> Union[int, float]:
+        """
+        Gets used car cost and calculate time used cars before disposal
+
+        :return:mileage before disposal
+        """
+        used_car_cost = Car.used_car_cost(self)
+        if self.engine == Car.DIESEL_ENGINE_TYPE:
+            time_before_disposal = used_car_cost / (Car.DECREASE_VALUE_DIESEL_PER_THOUSAND_KM / 1000)
+        else:
+            # For petrol car
+            time_before_disposal = used_car_cost / (Car.DECREASE_VALUE_PETROL_PER_THOUSAND_KM / 1000)
+        return round(time_before_disposal)
+
+
+class CarFabric:
+    """Class for car making"""
+
+    NUMBER_DIESEL_CAR = 3
+    NUMBER_NON_STANDARD_GAS_TANK = 5
+    VOLUME_STANDARD_GAS_TANK = 60
+    VOLUME_NON_STANDARD_GAS_TANK = 75
+
+    @classmethod
+    def produce_cars(cls, number_of_produce: int) -> list:
+        """
+        Gets parameters of the car and make car depending on the car number
+        :param number_of_produce: the number of cars to be created
+        :return:list of object class Car
+        """
+        cars = []
+        for i in range(number_of_produce):
+            if i % CarFabric.NUMBER_DIESEL_CAR == 0 and i % CarFabric.NUMBER_NON_STANDARD_GAS_TANK == 0:
+                cars.append(Car(Car.DIESEL_ENGINE_TYPE, CarFabric.VOLUME_NON_STANDARD_GAS_TANK))
+            elif i % CarFabric.NUMBER_DIESEL_CAR == 0 and i % CarFabric.NUMBER_NON_STANDARD_GAS_TANK != 0:
+                cars.append(Car(Car.DIESEL_ENGINE_TYPE, CarFabric.VOLUME_STANDARD_GAS_TANK))
+            elif i % CarFabric.NUMBER_DIESEL_CAR != 0 and i % CarFabric.NUMBER_NON_STANDARD_GAS_TANK == 0:
+                cars.append(Car(Car.PETROL_ENGINE_TYPE, CarFabric.VOLUME_NON_STANDARD_GAS_TANK))
+            else:
+                cars.append(Car(Car.PETROL_ENGINE_TYPE, CarFabric.VOLUME_STANDARD_GAS_TANK))
+        return cars
+
+
+class CarsHelperCalculationClass:
+    """Class for calculate another tasks"""
+
+    @staticmethod
+    def sorted_diesel_cars(all_cars):
+        """
+        Method for sorting cars by engine type
+        :return:list with diesel cars
+        """
+        diesel_car = []
+        for car in all_cars:
+            if car.engine == Car.DIESEL_ENGINE_TYPE:
+                diesel_car.append(car)
+        return diesel_car
+
+    @staticmethod
+    def sorted_petrol_cars(all_cars):
+        """
+        Method for sorting cars by engine type
+        :return:list with petrol cars
+        """
+        petrol_car = []
+        for car in all_cars:
+            if car.engine == Car.PETROL_ENGINE_TYPE:
+                petrol_car.append(car)
+        return petrol_car
```

Comment:
sorted_petrol_cars and sorted_diesel_cars are duplicates:
    @staticmethod
    def sort_cars_with_defined_engine_type(all_cars, engine_type):
        """
        Method for sorting cars by engine type
        :return:list with petrol cars
        """
        return [car for car in all_cars if car.engine == engine_type]